File: Practica-1/S2/CryptoVisionaryTUI/src/FactoriaAbstracta/CryptoCurrency.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package FactoriaAbstracta;


public class CryptoCurrency extends Thread{
    protected String token;
    protected Double valorActual=0.0;
    protected Double valorMinimo=0.0;
    protected Double valorMaximo=0.0;
    
    
    
    public void setToken(){
        
    }
    
    public void setValorActual(double nuevoValor){      
         if(nuevoValor < this.valorMinimo ){
               this.setValorMinimo(nuevoValor);
            }
            
            if(nuevoValor > valorMaximo){
                this.setValorMaximo(nuevoValor);
            }

      this.valorActual=nuevoValor; 
       
    }
    
    public void setValorMaximo(double nuevoValor){
        this.valorMaximo=nuevoValor;
    }
    
    public void setValorMinimo(double nuevoValor){
        this.valorMinimo=nuevoValor;
    }

    public String getToken() {
        return token;
    }

    public Double getValorActual() {
        return valorActual;
    }

    public Double getValorMinimo() {
        return valorMinimo;
    }

    public Double getValorMaximo() {
        return valorMaximo;
    }
    
    
    
    public void run(){
        
    }
    
    
}

File: Practica-2/App/crypto_visionary/lib/Cryptocurrency.dart
import 'package:flutter/material.dart';
import "dart:async";
import "package:threading/threading.dart";


class Cryptocurrency {
  String token;
  double capital;
  double valorActual;
  double valorMinimo;
  double valorMaximo;

  Cryptocurrency.empty() {
    this.token = "";
    this.capital = 0.0;
    this.valorActual = 0.0;
    this.valorMinimo = 0.0;
    this.valorMaximo = 0.0;
  }

  Cryptocurrency(String token, double capital, double valorActual, double valorMinimo, double valorMaximo) {
    this.token = token;
    this.capital = capital;
    this.valorActual = valorActual;
    this.valorMinimo = valorMinimo;
    this.valorMaximo = valorMaximo;
  }

   void setToken(String nuevoValor){
    this.token = nuevoValor;
  }

   void setValorActual(double nuevoValor){
    this.valorActual=nuevoValor;
  }

   void setValorMaximo(double nuevoValor){
    this.valorMaximo=nuevoValor;
  }

   void setValorMinimo(double nuevoValor){
    this.valorMinimo=nuevoValor;
  }

  void setCapital(double nuevoValor){
     this.capital=nuevoValor;
  }

   String getToken() {
    return token;
  }

   double getValorActual() {
    return valorActual;
  }

  double getValorMinimo() {
    return valorMinimo;
  }

  double getValorMaximo() {
    return valorMaximo;
  }

  double getCapital(){
     return this.capital;
  }


}